src/client.ts

```diff
@@ -2241,13 +2241,19 @@ export class MatrixClient extends TypedEventEmitter<EmittedEvents, ClientEventHa
             this.secretStorage,
             this.cryptoCallbacks,
             useIndexedDB ? RUST_SDK_STORE_PREFIX : null,
+            this.store,
         );
         rustCrypto.setSupportedVerificationMethods(this.verificationMethods);
 
         this.cryptoBackend = rustCrypto;
 
         // attach the event listeners needed by RustCrypto
         this.on(RoomMemberEvent.Membership, rustCrypto.onRoomMembership.bind(rustCrypto));
+        // listen only key verification request
+        // other events are processed during the /sync
+        this.on(RoomEvent.Timeline, (event) => {
+            rustCrypto.onKeyVerificationRequest(event);
+        });
```

Comment:
if we're going to feed all events into it, shouldn't it be called `onTimelineEvent` ?

Reply:
Except that we don't want to process all the timeline event. They are already processed in the sync response.

I did it in my fist impl and the event was processed twice.